src/mbgl/renderer/buckets/raster_bucket.hpp

```diff
@@ -1,22 +1,38 @@
 #pragma once
 
+#include <mbgl/gl/index_buffer.hpp>
+#include <mbgl/gl/texture.hpp>
+#include <mbgl/gl/vertex_buffer.hpp>
+#include <mbgl/programs/raster_program.hpp>
 #include <mbgl/renderer/bucket.hpp>
 #include <mbgl/util/image.hpp>
+#include <mbgl/util/mat4.hpp>
 #include <mbgl/util/optional.hpp>
-#include <mbgl/gl/texture.hpp>
 
 namespace mbgl {
 
 class RasterBucket : public Bucket {
 public:
     RasterBucket(UnassociatedImage&&);
+    RasterBucket(RasterBucket&&);
 
     void upload(gl::Context&) override;
     void render(Painter&, PaintParameters&, const RenderLayer&, const RenderTile&) override;
+    void render(Painter& painter,
+                PaintParameters& parameters,
+                const RenderLayer& layer,
+                const mat4& matrix);
     bool hasData() const override;
 
     UnassociatedImage image;
     optional<gl::Texture> texture;
+
+    gl::VertexVector<RasterLayoutVertex> vertices;
```

Comment:
Can you add a comment here explaining that these are used only when the layer has an image source?